sdk/src/ssi/credential.go

```diff
@@ -4,48 +4,61 @@ import (
 	"github.com/TBD54566975/ssi-sdk/credential"
 	"github.com/TBD54566975/ssi-sdk/credential/signing"
 	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
 	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/pkg/errors"
 )
 
-// SignVerifiableCredentialJWT takes in a key ID, key type, private key, and a verifiable credential
-// The keyID and KeyType are used to reconstruct a go-friendly private key to be used for signing
+// SignVerifiableCredentialJWT takes in a key ID, private JWK, and a verifiable credential.
+// The key ID and private JWK are used to reconstruct a go-friendly private key to be used for signing
 // the credential, which will be packaged as a JWT according to the VC-JWT 1.0 specification.
-// The resulting is returned as a string representation of a JWT.
-func SignVerifiableCredentialJWT(keyID, keyType string, privateKey, vcJSONBytes []byte) string {
-	privKey, err := crypto.BytesToPrivKey(privateKey, crypto.KeyType(keyType))
+// The resulting signed credential is returned as a string representation of a JWT.
+func SignVerifiableCredentialJWT(keyID string, privateJwk []byte, vcBytes []byte) (string, error) {
+	key, err := jwk.ParseKey(privateJwk)
```

Comment:
Is this because we don't expect clients to know what the keytype is and should we run with that assumption? I'm not knowledgable enough to answer that question but maybe @andresuribe87 @decentralgabe or @mistermoe knows

Reply:
KeyType can be derived from the JWK.

I understand this changes so that the clients can treat the JWKs in an opaque fashion.